mocha/specs/test.spec.js

```diff
@@ -1,152 +1,102 @@
-
-describe("Search for job", function () {
-  this.timeout(GLOBAL_TIMEOUT);
-
-  beforeEach(async () => {
-    await load();
-  });
-
-  describe("Careers page", () => {
-    it("should be opened", () => {
-      expect(element(by.css(".header__logo")).isDisplayed()).to.eventually.be.true;
-    });
-  });
-
-  describe("Search form", () => {
-    it("should be displayed", () => {
-      expect(element(by.css(".job-search__form")).isDisplayed()).to.eventually.be.true;
-    });
-
-    describe("Location Filter Box", () => {
-      beforeEach(async () => {
-        await selectLocation(testData);
-      });
-
-      it("should be able to select city and country", async () => {
-        expect(element(by.css(".select2-selection__rendered")).getText()).to.eventually.equal(testData.city);
-      });
+const data = require("../data/data.json");
+const CareerPage = require("../../page_objects/basePage");
+const SearchResultPage = require("../../page_objects/searchResultPage");
+const JobDescriptionPage = require("../../page_objects/jobDescriptionPage.js");
+const careerPage = new CareerPage();
+const searchResultPage = new SearchResultPage();
+const jobDescriptionPage = new JobDescriptionPage();
+
+data.forEach(testData => {
+  describe("Search for job", function () {
+    this.timeout(GLOBAL_TIMEOUT);
+
+    beforeEach(async () => {
+      careerPage.load();
+      careerPage.cookieClicker();
     });
 
-    describe("Department Filter Box", () => {
-      beforeEach(async () => {
-        await selectDepartment(testData);
-      });
-
-      it("should be able to select department ", () => {
-        expect(element(by.css(`li[data-value="${testData.department}"]`)).isEnabled()).to.eventually.be.true;
+    describe("Careers page", () => {
+      it("should be opened", () => {
+        expect(careerPage.logo.isDisplayed()).to.eventually.be.true;
       });
     });
 
-    describe("Searching", () => {
-      beforeEach(async () => {
-        await searchJob(testData);
+    describe("Search form", () => {
+      it("should be displayed", () => {
+        expect(careerPage.searchForm.isDisplayed()).to.eventually.be.true;
       });
 
-      it("should have proper job found ", async () => {
-        expect(element(by.css(`a.search-result__item-name[href*='.${jobFormat}']`)).isDisplayed()).to.eventually.be
-          .true;
-      });
+      describe("Location Filter Box", () => {
+        beforeEach(async () => {
+          await careerPage.selectLocation(testData.country, testData.city);
+        });
 
-      it.only("should have jobs with proper location", async () => {
-        const location = await element
-          .all(by.css(".search-result__item"))
-          .get(0)
-          .element(by.cssContainingText(".search-result__location", testData.country))
-          .getText();
-        expect(location).to.include(testData.country.toUpperCase());
+        it("should be able to select city and country", async () => {
```

Comment:
Could you please add the test data to the its? It would be useful for the debuging. It would be great if you would add them in all its.
e.g.: `should be able to select ${testData.city} city and ${testData.country} country`.

Reply:
done